subiquity/common/api/recoverable_error.py

```diff
@@ -15,4 +15,35 @@
 
 
 class RecoverableError(Exception):
-    """An error type that should not be treated as fatal."""
+    """An error type that can be treated as non-fatal when raised from a
+    request handler.
+
+    The intention of this class is to provide enough information about the
+    error to make it serializable into a Problem Details Object defined in RFC
+    9457 (Problem Details for HTTP APIs).
+
+    * The "code" attribute would be used as part of the "type" member of the
+    Problem Details Object. Maybe using the tag URI scheme:
+      tag:subiquity,2024-08-28:<code>
+    * The "title" attribute would be used as the "title" member of the Problem
+    Details Object.
+    * As for the message of the exception, it could be used as the "detail"
+    member of the Problem Details object."""
+
+    # If this exception is uncaught in a request handler, what should we do.
+    # By default we produce a crash report, but one can change set it to False.
+    produce_crash_report = True
```

Comment:
This also controls whether the client actually sees (or should treat) this as a recoverable error or not right? Also, having `RecoverableError` produce crash reports by default feels a little weird without the immediate context of the `--no-report-stroage-user-errors` flag changing this globally. Could we add some text to the comment about this behavior?